### PR TITLE
=pro #499 prepare for publishing Akka HTTP X, dont publish tests

### LINKED ISSUE #499: akka-http-tests should not be released

### PR DESCRIPTION
**Pull request by ktoso**
Resolves #499
Also some smaller build cleanup

### CODE REVIEW

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka-http/issues/437

File: build.sbt [hunk 1]
```diff
@@ -4,6 +4,7 @@ import akka._
 inThisBuild(Def.settings(
   organization := "com.typesafe.akka",
   organizationName := "Lightbend",
+  crossScalaVersions := Seq("2.11.8", "2.12.0"),
```
Comment (jonas) [inline build.sbt]: Note, `crossScalaVersions` is currently defined in [`project/Dependencies.scala`](https://github.com/akka/akka-http/blob/master/project/Dependencies.scala#L19) and added via `Dependencies.Versions` below. So if you want to define it here then I suggest the one in `Dependencies.Versions` is removed.

File: build.sbt [hunk 2]
```diff
@@ -44,6 +52,7 @@ lazy val root = Project(
     // Unidoc doesn't like macros
     unidocProjectExcludes := Seq(parsing)
   )
+  .settings(dontPublishSettings)
```
Comment (jonas) [inline build.sbt]: This should not be needed because of the `.enablePlugins(..., NoPublish)` above.

File: build.sbt [hunk 3]
```diff
@@ -30,12 +31,19 @@ inThisBuild(Def.settings(
   ),
   testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-v"),
   Dependencies.Versions,
-  Formatting.formatSettings
+  Formatting.formatSettings,
+  shellPrompt := { s => Project.extract(s).currentProject.id + " > " }
 ))
 
+val dontPublishSettings = Seq(
```
Comment (jonas) [inline build.sbt]: The [`NoPublish`](https://github.com/akka/akka-http/blob/master/project/Publish.scala#L8) plugin achieves the same thing, so this should not be needed.

File: build.sbt [hunk 4]
```diff
@@ -80,6 +89,7 @@ lazy val httpTestkit = project("akka-http-testkit")
 
 lazy val httpTests = project("akka-http-tests")
   .settings(Dependencies.httpTests)
+  .settings(dontPublishSettings)
```
Comment (jonas) [inline build.sbt]: With the plugin, this becomes `.enablePlugins(NoPublish)`

File: build.sbt (hunk 3 above)
Comment (ktoso) [inline build.sbt]: Heh, I see though the plugin is wrong it seems - needs to cancel publishSigned - the one we actually care about. Will change the PR

File: build.sbt (hunk 4 above)
Comment (ktoso) [inline build.sbt]: Thanks, fixing

File: build.sbt (hunk 3 above)
Comment (ktoso) [inline build.sbt]: Oh, there's no publish signed now... I see

**Pull request merged**